### ISSUE TITLE
[bug]: Extensions shouldn't have to tap `specialFeatures` constantly

### ISSUE THREAD

**Issue by zetlen** [magento/pwa-studio#2628]
Any extension with a React component, or any file that will go into the bundle, has to do this in its intercept: file

```js
targets.of('@magento/pwa-buildpack').tap(features => {
  features[targets.name] = {
    esModules: true,
    cssModules: true
  }
});
```

This is tedious, because most extensions WILL contain those types of modules. These flags should be set by default for every module.

The `specialFeatures` target is meant for extensions to flag that they contain certain file types which will be included into the build. Without explicitly including extensions into Webpack rule patterns, they wouldn't be parsed as ES Modules or CSS modules, because Webpack doesn't do that for all node_modules by default. If we made it do that, we'd have extremely slow builds, so we do need to explicitly include extensions.

But we can make the Webpack JS and CSS loaders include all code inside registered extensions and assume it's CSS and JS modules. Then, a lot of that boilerplate can go away.

Possible solutions Implementation notes

- This isn't completely trivial because we still need to keep those rules false "by default", for all the hundreds of Node modules. To have them on "by default" you would need to refactor a bit.

- BuildBus should have an accessor for the list of all active extensions; it can gather that from its call to `pertain`.

- In `getModuleRules()`, the `hasFlag(foo)` method should return all active extension directories by default, only removing one if it has explicitly tapped `specialFeatures` to set `features[targets.name] = { foo: false }`

- The UpwardIncludePlugin should no longer error if an `upward.yml` isn't found, it should just move on

**Please let us know what packages this bug is in regards to:**
- [ ] `venia-concept`
- [ ] `venia-ui`
- [x] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

**Comment by awilcoxa** [comment]
Created PWA-866

**Comment by larsroettig** [comment]
@magento I am working on this

**Comment by larsroettig** [comment]
@zetlen `runPhase` already know about what are is active PWA Studio extensions. 

https://github.com/magento/pwa-studio/blob/6da17c185871bbc1260646be04dfc7a1d6e76435/packages/pwa-buildpack/lib/BuildBus/BuildBus.js#L179-L202

**Comment by tjwiebell** [comment]
Addressed by #2765, going to close this out.
